<a href="https://colab.research.google.com/github/dlhegend/-TS_Academy_Capstone_Project/blob/main/notebooks/Gift_M_TSAcademy_Cap_Project.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [10]:
import pandas as pd
import matplotlib.pyplot as plt


In [11]:
df = pd.read_csv('/content/NGX_ASI_Index_Price_Data.csv')

In [12]:
df.head()

date  avr_monthly_kiloprice
0  2012-Jan                  20.80
1  2012-Feb                  20.52
2  2012-Mar                  20.92
3  2012-Apr                  21.27
4  2012-May                  22.41

In [13]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 137 entries, 0 to 136
Data columns (total 2 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   date                   137 non-null    object 
 1   avr_monthly_kiloprice  137 non-null    float64
dtypes: float64(1), object(1)
memory usage: 2.3+ KB


The data set has 137 records (indexed from 0-136)
2 Columns are contained in the dataset.
No missing Values
Data types are float64 and objects.

Objects for date column is not good for timeseries data, as pandas is not reading the column as date but as text, this need to be converted to sute the dataset type.

In [14]:
#convert date to time

df['date'] = pd.to_datetime(df['date'])


/tmp/ipython-input-3571739026.py:3: UserWarning: Could not infer format, so each element will be parsed individually, falling back to `dateutil`. To ensure parsing is consistent and as-expected, please specify a format.
  df['date'] = pd.to_datetime(df['date'])


In [15]:
#set index

df.set_index('date', inplace=True)


In [16]:
#checking info again
df.info()

<class 'pandas.core.frame.DataFrame'>
DatetimeIndex: 137 entries, 2012-01-01 to 2023-05-01
Data columns (total 1 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   avr_monthly_kiloprice  137 non-null    float64
dtypes: float64(1)
memory usage: 2.1 KB


In [17]:
df.describe()

avr_monthly_kiloprice
count             137.000000
mean               34.301241
std                 8.428547
min                20.520000
25%                27.430000
50%                33.350000
75%                39.270000
max                55.080000

The Standard Deviation is much smaller than the mean (at about 25%) indicating moderate variability around the mean.

In [18]:
#comfirming missing values

df.isna().sum()

avr_monthly_kiloprice    0
dtype: int64

No Missing Values